yodeploy/repository.py

```diff
@@ -97,11 +102,29 @@ def get_metadata(self, path):
         with open(meta_fn) as f:
             return json.load(f)
 
-    def put(self, path, fp, metadata=None):
-        '''
-        Store a file (fp).
+    def put(self, path, data, metadata=None):
+        """
+        Store a file (fp), stream, unicode string, or byte string.
```

Comment:
If you're saying `data` can be a file pointer, maybe we should be more specific?

This also seems rather unnecessarily flexible, wouldn't file objects and byte strings be enough? Supporting the unicode case brings in a bit of complexity.